### PR TITLE
Offerings cleanup

### PR DESCRIPTION
**Pull request by beylmk**
While reading through the getOfferings flow, I found some of it a little tricky to read. This PR has no functionality changes, just some renames/refactors to make things more readable as I prep to update getOfferings for BC5.

### CODE REVIEW

File: common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            if (it.availablePackages.isEmpty()) {
+                warnLog(OfferingStrings.OFFERING_EMPTY.format(it.identifier))
+            }
+        }
+    }
+
+    return Offerings(offerings[currentOfferingID], offerings)
+}
+
+fun JSONObject.createOffering(productsById: Map<String, StoreProduct>): Offering? {
```
Comment (beylmk) [inline common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt:33]: only change besides pulling into a new file is renaming the parameters from `products` -> `productsById` and pulling out some intermediate variables in the json parsing

File: purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt [hunk 2]
```diff
@@ -1237,34 +1237,35 @@ class Purchases internal constructor(
     }
 
     private fun getSkuDetails(
-        skus: Set<String>,
+        productIds: Set<String>,
```
Comment (beylmk) [inline purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt:1240]: this rename is kinda the only reason i wouldn't merge this right into main...changing from skus makes this more generic for both BC4 and BC5

File: purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt [hunk 3]
```diff
@@ -1237,34 +1237,35 @@ class Purchases internal constructor(
...
         onError: (PurchasesError) -> Unit
     ) {
         billing.querySkuDetailsAsync(
             ProductType.SUBS,
-            skus,
-            { subscriptionsSKUDetails ->
-                val detailsByID = HashMap<String, StoreProduct>()
-                val inAPPSkus =
-                    skus - subscriptionsSKUDetails
-                        .map { details -> details.sku to details }
```
Comment (beylmk) [inline purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt:1251]: i personally found the chaining here a little hard to follow, so i pulled them out and made some more variables. open to undoing if it doesn't seem worth the tradeoff

File: purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt [hunk 4]
```diff
@@ -1237,34 +1237,35 @@ class Purchases internal constructor(
...
-
-                if (inAPPSkus.isNotEmpty()) {
+            productIds,
+            { subscriptionProducts ->
+                val productsById = HashMap<String, StoreProduct>()
+
+                val subscriptionProductsById = subscriptionProducts.map { subProduct -> subProduct.sku to subProduct }
+                productsById.putAll(subscriptionProductsById)
+
+                val subscriptionIds = subscriptionProductsById.map { subProductById -> subProductById.first }.toSet()
```
Comment (tonidero) [inline purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt:1253]: Hmm I think this can be simply `subscriptionProductsById.keys`?

**Review by tonidero — APPROVED** [review]
Looks great to me!
